src/Sulu/Component/Content/Document/SynchronizationManager.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document;
+
+use Sulu\Bundle\ContentBundle\Document\RouteDocument;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentInspector;use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentManagerRegistry;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\PropertyEncoder;
+use Sulu\Component\Content\Document\Behavior\ResourceSegmentBehavior;
+use Sulu\Component\Content\Document\Behavior\SynchronizeBehavior;
+use Sulu\Component\Content\Document\Syncronization\DocumentRegistrator;
+use Sulu\Component\DocumentManager\Behavior\Mapping\LocaleBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerRegistryInterface;
+
+/**
+ * The synchronization manager handles the synchronization of documents
+ * from the DEFAULT document manger to the PUBLISH document manager.
+ *
+ * NOTE: In the future multiple document managers may be supported.
+ *
+ * TODO: Explcitly inject publish and default document managers?
+ */
+class SynchronizationManager
+{
+    /**
+     * @var DocumentManagerRegistryInterface
+     */
+    private $registry;
+
+    /**
+     * @var PropertyEncoder
+     */
+    private $encoder;
+
+    /**
+     * @var string
+     */
+    private $publishManagerName;
+
+    /**
+     * @var DocumentRegistrator
+     */
+    private $registrator;
+
+    public function __construct(
+        DocumentManagerRegistry $registry,
+        PropertyEncoder $encoder,
+        $publishManagerName,
+        $registrator = null
+    ) {
+        $this->registry = $registry;
+        $this->publishManagerName = $publishManagerName;
+        $this->encoder = $encoder;
+        $this->registrator = $registrator ?: new DocumentRegistrator(
+            $registry->getManager(),
+            $registry->getManager($this->publishManagerName)
+        );
+    }
+
+    /**
+     * Return the publish document manager (PDM).
+     *
+     * This should be the only class that is aware of the PDM name. By having
+     * this method we can be sure that whatever the PDM is, the PDM is always
+     * the PDM.
+     *
+     * NOTE: This is used only by the synchronization subscriber in order
+     *       to "flush" the PDM.
+     *
+     * @return DocumentManagerInterface
+     */
+    public function getPublishDocumentManager()
+    {
+        return $this->registry->getManager($this->publishManagerName);
+    }
+
+    /**
+     * Synchronize a document and any other documents which are
+     * associated with it and should also be published.
+     *
+     * All document managers involved will be FLUSHED after
+     * the operation has completed.
+     *
+     * @param SynchronizeBehavior $document
+     */
+    public function synchronizeFull(SynchronizeBehavior $document, array $options = [])
```

Comment:
It is already removed in dm_sync_cascade branch.

On Mon, Apr 18, 2016 at 12:57:21AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Content/Document/SynchronizationManager.php:
> 
>  > +     */
>  > +    public function getPublishDocumentManager()
>  > +    {
>  > +        return $this->registry->getManager($this->publishManagerName);
>  > +    }
>  > +
>  > +    /**
>  > +     * Synchronize a document and any other documents which are
>  > +     * associated with it and should also be published.
>  > +     *
>  > +     * All document managers involved will be FLUSHED after
>  > +     * the operation has completed.
>  > +     *
>  > +     * @param SynchronizeBehavior $document
>  > +     */
>  > +    public function synchronizeFull(SynchronizeBehavior $document, array $options = [])
> 
>    When we solve this issue as discussed last week (so that we only have to
>    update two route nodes, the new one, and the old one not being a history),
>    then this method might not be necessary at all, right?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/dantleech/sulu/pull/2#discussion_r60018003
>    2. https://github.com/dantleech/sulu/pull/2/files/f78a7ede8c61309e7789a237e25fed93d13568e8#r60018003
>    3. https://github.com/dantleech/sulu/pull/2/files/f78a7ede8c61309e7789a237e25fed93d13568e8#r60018003